src/sentry/db/postgres/schema.py

```diff
@@ -97,6 +112,71 @@ def remove_field(self, model, field):
+            statements.extend(Statement(s, **sql.parts) for s in sql.template)
+        else:
+            statements.append(sql)
+        for statement in statements:
+            if isinstance(statement, PGLock):
+                use_timeouts = statement.use_timeouts
+                disable_statement_timeout = statement.disable_statement_timeout
+                statement = statement.sql
+            elif isinstance(statement, Statement) and isinstance(statement.template, PGLock):
+                use_timeouts = statement.template.use_timeouts
+                disable_statement_timeout = statement.template.disable_statement_timeout
+                statement = Statement(statement.template.sql, **statement.parts)
+            else:
+                use_timeouts = False
+                disable_statement_timeout = False
+
+            if use_timeouts:
+                with self._set_operation_timeout(self.STATEMENT_TIMEOUT, self.LOCK_TIMEOUT):
+                    PostgresDatabaseSchemaEditor.execute(self, statement, params)
+            elif self.LOCK_TIMEOUT_FORCE:
+                with self._set_operation_timeout(lock_timeout=self.LOCK_TIMEOUT):
+                    PostgresDatabaseSchemaEditor.execute(self, statement, params)
+            elif disable_statement_timeout and self.FLEXIBLE_STATEMENT_TIMEOUT:
+                with self._set_operation_timeout(self.ZERO_TIMEOUT):
+                    PostgresDatabaseSchemaEditor.execute(self, statement, params)
+            else:
+                PostgresDatabaseSchemaEditor.execute(self, statement, params)
+
+    @contextmanager
+    def _set_operation_timeout(self, statement_timeout=None, lock_timeout=None):
+        if self.collect_sql:
+            previous_statement_timeout = self.ZERO_TIMEOUT
+            previous_lock_timeout = self.ZERO_TIMEOUT
+        else:
+            with self.connection.cursor() as cursor:
+                cursor.execute(self.sql_get_statement_timeout)
+                (previous_statement_timeout,) = cursor.fetchone()
+                cursor.execute(self.sql_get_lock_timeout)
+                (previous_lock_timeout,) = cursor.fetchone()
+        if statement_timeout is not None:
+            PostgresDatabaseSchemaEditor.execute(
+                self, self.sql_set_statement_timeout % {"statement_timeout": statement_timeout}
+            )
+        if lock_timeout is not None:
+            PostgresDatabaseSchemaEditor.execute(
+                self, self.sql_set_lock_timeout % {"lock_timeout": lock_timeout}
+            )
+        yield
+        if statement_timeout is not None:
+            PostgresDatabaseSchemaEditor.execute(
+                self,
+                self.sql_set_statement_timeout % {"statement_timeout": previous_statement_timeout},
+            )
+        if lock_timeout is not None:
+            PostgresDatabaseSchemaEditor.execute(
+                self, self.sql_set_lock_timeout % {"lock_timeout": previous_lock_timeout}
+            )
```

Comment:
Copied from https://github.com/tbicr/django-pg-zero-downtime-migrations/blob/a96fb4dea8361857ca24b05e84a9ab6f612a3bb6/django_zero_downtime_migrations/backends/postgres/schema.py#L284-L335 and modified